Add kmcp classifier (#312)
nf-core/taxprofiler: docs/output.md

Normally `*_combined_reports.txt` is the most useful file for downstream analyses, but the per sample `.out` file can provide additional more specific information. By default, nf-core/taxprofiler is providing a column describing NCBI taxonomic ID as this is used in the taxpasta step. You can disable this column by activating the argument `--motus_remove_ncbi_ids`.
You will receive the relative abundance instead of read counts if you provide the argument `--motus_use_relative_abundance`.

### kmcp

[kmcp](https://github.com/shenwei356/kmcp) utilizes genome coverage information by splitting the reference genomes into chunks and stores k-mers in a modified and optimized COBS index for fast alignment-free sequence searching. KMCP combines k-mer similarity and genome coverage information to reduce the false positive rate of k-mer-based taxonomic classification and profiling methods.

> 

Reviewer: According to the original paper, it seems that `KMCP` is more resource-efficient. It would be very interesting to compare it with the other two k-mer based methods, `Kraken2` and `Ganon`, in our clinical cases.

Author: Agree!

Normally `*_combined_reports.txt` is the most useful file for downstream analyses, but the per sample `.out` file can provide additional more specific information. By default, nf-core/taxprofiler is providing a column describing NCBI taxonomic ID as this is used in the taxpasta step. You can disable this column by activating the argument `--motus_remove_ncbi_ids`.
You will receive the relative abundance instead of read counts if you provide the argument `--motus_use_relative_abundance`.

### kmcp

[kmcp](https://github.com/shenwei356/kmcp) utilizes genome coverage information by splitting the reference genomes into chunks and stores k-mers in a modified and optimized COBS index for fast alignment-free sequence searching. KMCP combines k-mer similarity and genome coverage information to reduce the false positive rate of k-mer-based taxonomic classification and profiling methods.

<summary>Output files</summary>

> - `kmcp`

Author: Done!

Normally `*_combined_reports.txt` is the most useful file for downstream analyses, but the per sample `.out` file can provide additional more specific information. By default, nf-core/taxprofiler is providing a column describing NCBI taxonomic ID as this is used in the taxpasta step. You can disable this column by activating the argument `--motus_remove_ncbi_ids`.
You will receive the relative abundance instead of read counts if you provide the argument `--motus_use_relative_abundance`.

### kmcp

> [kmcp](https://github.com/shenwei356/kmcp) utilizes genome coverage information by splitting the reference genomes into chunks and stores k-mers in a modified and optimized COBS index for fast alignment-free sequence searching. KMCP combines k-mer similarity and genome coverage information to reduce the false positive rate of k-mer-based taxonomic classification and profiling methods.

Reviewer 2: ```suggestion
[KMCP](https://github.com/shenwei356/kmcp) utilises genome coverage information by splitting the reference genomes into chunks and stores k-mers in a modified and optimised COBS index for fast alignment-free sequence searching. KMCP combines k-mer similarity and genome coverage information to reduce the false positive rate of k-mer-based taxonomic classification and profiling methods.
```

<details markdown="1">
<summary>Output files</summary>

- `kmcp`

  - `<db_name>/`

>     - `<sample_id>.gz`: output of `kmcp_search` containing search sequences against a database.

Reviewer 2: On a per-read basis?

Author: It is per-sample basis if I understand your question correctly.

Reviewer 2: Ok, I don't understand the difference between the two files then based on the current description

Author: I updated the PR now.

    - `<sample_id>_kmcp.profile`: output of `kmcp_profile` containing the taxonomic profile from search results.

    </details>

> 

Author: Updated the PR

You will receive the relative abundance instead of read counts if you provide the argument `--motus_use_relative_abundance`.

### KMCP

[KMCP](https://github.com/shenwei356/kmcp) utilises genome coverage information by splitting the reference genomes into chunks and stores k-mers in a modified and optimised COBS index for fast alignment-free sequence searching. KMCP combines k-mer similarity and genome coverage information to reduce the false positive rate of k-mer-based taxonomic classification and profiling methods.

<details markdown="1">
<summary>Output files</summary>

- `kmcp`

  - `<db_name>/`

    - `<sample_id>.gz`: output of `kmcp_search` containing search sequences against a database in tab-delimited format with 15 columns.
    - `<sample_id>_kmcp.profile`: output of `kmcp_profile` containing the taxonomic profile from search results.

  </details>

>   You will receive the `<sample_id>.gz` file if you supply `--kmcp_save_search`. Please note that there is no taxonomic label assignment in this output file.

Reviewer 2: ```suggestion
You will receive the `<sample_id>.gz` file if you supply `--kmcp_save_search`. Please note that there is no taxonomic label assignment in this output file.
```

> The main taxonomic classification file from kmcp is the `*kmcp.profile` which is also used by the taxpasta step.

Reviewer 2: ```suggestion
The main taxonomic classification file from KMCP is the `*kmcp.profile` which is also used by the taxpasta step.
```
